uncertainties/core.py

```diff
@@ -65,139 +66,151 @@
 
 try:
     import numpy
+
+    __all__ += [
+        "correlated_values",
+        "correlated_values_norm",
+        "correlation_matrix",
+    ]
 except ImportError:
-    pass
-else:
-    # Entering variables as a block of correlated values.  Only available
-    # if NumPy is installed.
+    msg = "Unable to import numpy. Some functionality will be unavailable"
+    warnings.warn(msg)
+    numpy = None
 
-    #! It would be possible to dispense with NumPy, but a routine should be
-    # written for obtaining the eigenvectors of a symmetric matrix.  See
-    # for instance Numerical Recipes: (1) reduction to tri-diagonal
-    # [Givens or Householder]; (2) QR / QL decomposition.
 
-    def correlated_values(nom_values, covariance_mat, tags=None):
-        """
-        Return numbers with uncertainties (AffineScalarFunc objects)
-        that correctly reproduce the given covariance matrix, and have
-        the given (float) values as their nominal value.
+def correlated_values(nom_values, covariance_mat, tags=None):
```

Comment:
I think there are two ways we can change things here:

1. Put some check at the top of the function to make sure that `numpy` was imported correctly and raise an exception if not.
2. Conditionally define the functions like was done before but in the else case instead of not defining the functions define functions with the same name and signature but with bodies that just raise an exception about numpy missing.

The first option seems simpler. It just has that extra check at the start of every call but that should be okay.

For the exception I think it would be a custom exception, or maybe `ModuleNotFoundError`, though it feels a little weird to raise an `ImportError` during a function call.

Minor point -- maybe the `try` should be excepting `ModuleNotFoundError` instead of `ImportError`, just in case `numpy` is installed but there is some other problem with importing it? Right now, Uncertainties hides that other import error from the user just like if numpy were not installed.